lib/active_merchant/billing/gateways/checkout_v2.rb

```diff
@@ -459,37 +452,48 @@ def access_token_url
       end
 
       def setup_access_token
-        request = 'grant_type=client_credentials'
-        begin
-          raw_response = ssl_post(access_token_url, request, access_token_header)
-        rescue ResponseError => e
-          raise OAuthResponseError.new(e)
-        else
-          response = parse(raw_response)
+        response = parse(ssl_post(access_token_url, 'grant_type=client_credentials', access_token_header))
+        @options[:access_token] = response['access_token']
+        @options[:expires] = response['expires_in'] && response['expires_in'] > 0 && (DateTime.now + (response['expires_in'] - 120).seconds).strftime('%Q').to_i
 
-          if (access_token = response['access_token'])
-            access_token
-          else
-            raise OAuthResponseError.new(response)
-          end
-        end
+        Response.new(
+          response['access_token'].present?,
+          message_from(true, response, {}),
+          response.merge({ expires: @options[:expires] }),
+          test: test?,
+          error_code: error_code_from(false, response, {})
+        )
+      rescue ResponseError => e
+        raise OAuthResponseError.new(e)
       end
 
-      def commit(action, post, options, authorization = nil, method = :post)
+      def access_token_valid?
+        @options[:access_token].present? && @options[:expires] && @options[:expires].to_i > DateTime.now.strftime('%Q').to_i
```

Comment:
Since you've defaulted expires to be 0 i don't think the extract @options[:expires] is necessary

Reply:
make sense